src/components/StorageOverview/Alerts/Alerts.js

```diff
@@ -1,39 +1,94 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Row, Col } from 'patternfly-react';
 
 import {
   DashboardCard,
   DashboardCardBody,
   DashboardCardHeader,
   DashboardCardTitle,
 } from '../../Dashboard/DashboardCard';
+import { DashboardCardActionsBody } from '../../Dashboard/DashboardCard/DashboardCardActionsBody';
+import { Dropdown } from '../../Form/Dropdown';
+
 import { StorageOverviewContext } from '../StorageOverviewContext';
 import { AlertsBody } from '../../Dashboard/Alert/AlertsBody';
 import { filterAlerts } from './utils';
 import { AlertItem } from '../../Dashboard/Alert/AlertItem';
+import { getAlertSeverity } from '../../../selectors/prometheus/alerts';
+import { ALL, WARNING, CRITICAL } from './strings';
+
+const alertsFilterOptions = [ALL, CRITICAL, WARNING];
 
-export const Alerts = ({ alertsResponse, className }) => {
-  if (!Array.isArray(alertsResponse)) {
-    return null;
+export class Alerts extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: alertsFilterOptions[0],
+    };
   }
 
-  const alerts = filterAlerts(alertsResponse);
-
-  return alerts.length > 0 ? (
-    <DashboardCard className={className}>
-      <DashboardCardHeader>
-        <DashboardCardTitle>Alerts</DashboardCardTitle>
-      </DashboardCardHeader>
-      <DashboardCardBody>
-        <AlertsBody>
-          {alerts.map((alert, index) => (
-            <AlertItem key={`alert-${index}`} alert={alert} />
-          ))}
-        </AlertsBody>
-      </DashboardCardBody>
-    </DashboardCard>
-  ) : null;
-};
+  onDropdownOptionChange = newVal => this.setState({ type: newVal }, this.getAlerts);
+
+  filterAlertsBy = (allAlerts, filter) => {
+    const filteredAlerts = allAlerts.filter(alert => getAlertSeverity(alert) === filter);
+    /* handles the case when no alerts of a particular filter are present */
+    return filteredAlerts.length !== 0 ? filteredAlerts : allAlerts;
```

Comment:
IMO we should show zero alerts if the user wishes the filter to be that way.

Reply:
yeah even I tried that but the issue is when no alerts i.e [] array is passed then the entire alert component is gone since inside render we have this condition `return alerts.length > 0` and `null` is returned.
So if user wants to select any other filter, he wont be able to do that.

Reply:
We could change that condition so that alerts card is visible when some alert exists, i.e. `allAlerts.length > 0`

Reply:
@suomiy is it looks good now